### ISSUE TITLE
Failing Test : [sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] volumeIO should write files of various sizes, verify size, validate content [Slow]

### ISSUE THREAD

**Issue by AishSundar** [kubernetes/kubernetes#71183]
**Which jobs are failing**:
* [soak-gci-gce-1.13](https://k8s-testgrid.appspot.com/sig-release-1.13-all#soak-gci-gce-1.13)

**Which test(s) are failing**:
[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] volumeIO should write files of various sizes, verify size, validate content [Slow]

**Since when has it been failing**:
11/13

**Testgrid link**:
https://k8s-testgrid.appspot.com/sig-release-1.13-all#soak-gci-gce-1.13

**Reason for failure**:
https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-soak-gci-gce-beta/484
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/volume_io.go:152
Expected error:
    <*errors.errorString | 0xc000ac6330>: {
        s: "size of file /opt/volume/e2e-tests-volumes-jzwmx/hostPath_io_test_e2e-tests-volumes-jzwmx-1048576 is 2097152, expected 1048576",
    }
    size of file /opt/volume/e2e-tests-volumes-jzwmx/hostPath_io_test_e2e-tests-volumes-jzwmx-1048576 is 2097152, expected 1048576
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/volume_io.go:157
```

**Anything else we need to know**:
This is consistently failing in 1.13 branch and is release blocking.

/kind failing-test
/sig storage
/priority critical-urgent
/milestone v1.13
/assign @msau42 

/cc @saad-ali @jberkus @mariantalla @mortent

**Comment by msau42** [comment]
There's a few problems:
1. hostPathDriver.GetVolumeSource returns "/tmp" for everything.  This means all tests will share the same directory.  This should be probably use some unique directory instead
2. The volume IO test generates a unique mountPath directory, however that doesn't mean a unique directory is actually created on the host.  Ie, a mountPath of "/a/b/c" on a hostpath volume at "/tmp" doesn't mean directory "/tmp/a/b/c" is created.
3. The io test file name is not unique
4. It seems like the IO test is not cleaning up the files it creates, so the files are getting reused on subsequent tests due to the non-unique naming issues.

**Comment by msau42** [comment]
The other thing I can't explain is why it's failing only now

**Comment by msau42** [comment]
@kubernetes/sig-storage-bugs

**Comment by msau42** [comment]
Fix is here: https://github.com/kubernetes/kubernetes/pull/71235

**Comment by AishSundar** [comment]
As per discussion in 1.13 burndown this morning, this is a test job issue and its not release blocking at this point. So we can take this fix in 1.14 as opposed now during code freeze

@jberkus let us know if you feel otherwise
